### PR TITLE
feat: add docker and compose configs #64

### PR DESCRIPTION
**Pull request by Shrest4647**
## Related Issue

Issue No. #64 
## Description

Implemented and configured the application containerization using Dockerfile and set up automation using compose for easier setup. Also the docs has been updated for the same.

Docker/Containers are the new standard for sharing development configuration as well as for easier deployment. It enables user to the share configuration between  dev, test and production environments easily.

### CODE REVIEW

**Comment by Shrest4647** [comment]
@Sanchitbajaj02 Could you clarify what this Vercel check is and how it can be passed successfully?

**Comment by Sanchitbajaj02** [comment]
You can ignore the vercel check as it deploy the changes you've made on vercel but you should have to pass the PR title checker

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM node:18.17.0-alpine
```
Comment (Sanchitbajaj02) [inline Dockerfile:1]: Don't you think it will too latest? can we downgrade to node 16

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+WORKDIR /app
+
+# Update npm
+RUN npm install -g npm@latest
+
+# Install app dependencies
+COPY ["package.json", "package-lock.json*", "yarn.lock*", "./"]
```
Comment (Sanchitbajaj02) [inline Dockerfile:9]: the project is solemnly dependent on yarn so you can remove package-lock.json

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: '3'
```
Comment (Sanchitbajaj02) [inline docker-compose.dev.yml:1]: What's the difference between docker-compose.yml and docker-compose.dev.yml?

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Shrest4647) [reply Dockerfile:1]: Sure should node 16 be okay?

File: Dockerfile (hunk 2 above)
Comment (Shrest4647) [reply Dockerfile:9]: Okay I will remove the package-lock dependency

File: docker-compose.dev.yml (hunk 3 above)
Comment (Shrest4647) [reply docker-compose.dev.yml:1]: When using configuration from docker-compose.dev.yml, next is started in development mode and supports hot reloading the changes through volumes. However, the docker-compose.yml file has settings tuned for deployment and serves the built and optimized server.

Comment (Sanchitbajaj02) [reply docker-compose.dev.yml:1]: ok

File: Dockerfile (hunk 1 above)
Comment (Shrest4647) [reply Dockerfile]: Done node image set to v.16.20.2

File: Dockerfile (hunk 2 above)
Comment (Shrest4647) [reply Dockerfile]: package-lock.json removed from the dependency list

**Review by Sanchitbajaj02 — APPROVED** [review]

**Pull request merged**